### PR TITLE
Move ARM cmdlets from Resources to Resources.Rest proj

### PR DESCRIPTION
**Pull request by vivsriaus**
This is a massive change list that impacts the core ARM cmdlets like ResourceGroups, deployments, features and providers. We're moving these cmdlets which are currently using the old HYAK client into the Commands.Rest project, using the new autorest SDK client. More updates will follow shortly

### CODE REVIEW

**Comment by vivsriaus** [comment]
@azuresdkci test this please

**Comment by vivsriaus** [comment]
@azuresdkci retest this please

**Comment by markcowl** [comment]
@vivsriaus Where do we sit with this PR?

**Comment by vivsriaus** [comment]
@markcowl I need help in debugging the latest test failures. I'm on-call for ARM this week, so I'm not able to look at this currently, but if you can help me why I keep getting [this ](http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/769/artifact/src/Publish/TestResults/AzurePowershellAllTestResults.html#failures) error, that'd be helpful.

**Comment by markcowl** [comment]
@azuresdkci retest this please

**Comment by vivsriaus** [comment]
Here is the on-demand job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/785/

Thanks @TianoMS

**Comment by vivsriaus** [comment]
@azuresdkci retest this please

File: src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/AzureRM.Resources.ps1 [hunk 1]
```diff
@@ -0,0 +1,158 @@
+﻿function Get-AzureRmResourceGroup
+{
+  [CmdletBinding()]
+  param(
+    [string] [Parameter(Position=0, ValueFromPipelineByPropertyName=$true)] [alias("ResourceGroupName")] $Name,
+    [string] [Parameter(Position=1, ValueFromPipelineByPropertyName=$true)] $Location,
+	[string] [Parameter(ValueFromPipelineByPropertyName=$true)] $Id,
```
Comment (gandhiniraj) [inline src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/AzureRM.Resources.ps1]: nit: aligning

File: src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/AzureRM.Resources.ps1 [hunk 2]
```diff
@@ -0,0 +1,158 @@
+﻿function Get-AzureRmResourceGroup
```
Comment (gandhiniraj) [inline src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/AzureRM.Resources.ps1]: nit: I assume VS doesn't do a good job on aligning PS, but we should maintain that.

File: src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/AzureRM.Resources.ps1 (hunk 1 above)
Comment (gandhiniraj) [inline src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/AzureRM.Resources.ps1]: nit: e.g. aligning here

File: src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/AzureRM.Resources.ps1 [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  PROCESS {
+    if($Name -eq $null) {
+	  $getTask = $client.ResourceGroups.ListAsync($null, [System.Threading.CancellationToken]::None)
+	  $rg = $getTask.Result
+	  $resourceGroup = List-ResourceGroup
+	  Write-Output $resourceGroup
+	} else {
+	  $getTask = $client.ResourceGroups.GetAsync($Name, [System.Threading.CancellationToken]::None)
+	  $rg = $getTask.Result
+	  $resourceGroup = Get-ResourceGroup $Name $Location $rg.ResourceGroup.Id
```
Comment (gandhiniraj) [inline src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/AzureRM.Resources.ps1]: Do we need to also check if $Location is null?

File: src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/AzureRM.Resources.ps1 (hunk 1 above)
Comment (gandhiniraj) [inline src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/AzureRM.Resources.ps1]: I don't see $Id is being used below, do we still need get it?

File: src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/AzureRM.Resources.ps1 [hunk 4]
```diff
@@ -0,0 +1,158 @@
+﻿function Get-AzureRmResourceGroup
+{
+  [CmdletBinding()]
+  param(
```
Comment (gandhiniraj) [inline src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/AzureRM.Resources.ps1]: nit: PARAM to be in sync with BEGIN and PROCESS

File: src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/AzureRM.Resources.ps1 [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+	[string] [Parameter(ValueFromPipelineByPropertyName=$true)] $Tags,
+	[switch] $Force)
+  BEGIN { 
+    $context = Get-Context
+	$client = Get-ResourcesClient $context
+  }
+  PROCESS {
+    $createParms = New-Object -Type Microsoft.Azure.Management.Resources.Models.ResourceGroup
+	$createParms.Location = $Location
+	#$createParms.Tags = $Tags
```
Comment (gandhiniraj) [inline src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/AzureRM.Resources.ps1]: nit: remove commented code

File: src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/AzureRM.Resources.ps1 [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+    [string] [Parameter(Position=0, ValueFromPipelineByPropertyName=$true)] [alias("ResourceGroupName")] $Name,
+    [string] [Parameter(Position=1, ValueFromPipelineByPropertyName=$true)] $Location,
+	[string] [Parameter(ValueFromPipelineByPropertyName=$true)] $Tags,
+	[switch] $Force)
+  BEGIN { 
+    $context = Get-Context
+	$client = Get-ResourcesClient $context
+  }
+  PROCESS {
+    $createParms = New-Object -Type Microsoft.Azure.Management.Resources.Models.ResourceGroup
```
Comment (gandhiniraj) [inline src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/AzureRM.Resources.ps1]: createParams?

**Comment by vivsriaus** [comment]
@gandhiniraj Addressed your feedback. Thanks

**Comment by vivsriaus** [comment]
Latest on demand job here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/787/

**Comment by hovsepm** [comment]
@vivsriaus please resolve merge conflicts

**Comment by vivsriaus** [comment]
@hovsepm done

**Comment by markcowl** [comment]
on demand job after all changes: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/799/

**Comment by markcowl** [comment]
signing job here:  http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/432/

**Comment by hovsepm** [comment]
@vivsriaus the failures in on-demand job is fixed by Batch team and are merged. Please pull from upstream and restart the on-demand after that.

**Comment by markcowl** [comment]
@vivciraus - I think the issues are related to merge problems- can you pull the latest from the dev branch?

**Comment by vivsriaus** [comment]
@hovsepm @markcowl Here is the new on-demand job. Hopefully, this should do it: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/815/

**Comment by vivsriaus** [comment]
@markcowl @hovsepm More changes were merged from upstream dev, so here is the new on-demand job (the previous one succeeded, FWIW): http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/816/

**Comment by vivsriaus** [comment]
New on demand job here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/821/

**Comment by hovsepm** [comment]
sign job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/436/

**Comment by vivsriaus** [comment]
Latest on demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/827/

**Comment by hovsepm** [comment]
Failing.playlist  and NewFailing.playlist should not be checked in

**Comment by hovsepm** [comment]
@ src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Components\ResourceIdentifier.cs, Line 146: private void AppendIfNotNull(ref StringBuilder resourceId, string format, string value) -> no need to use ref for StringBuilder, you are not changing reference of resourceId in the body anyhow.

**Comment by hovsepm** [comment]
@vivsriaus Since this is a huge change in the existing code please include ChangeLog.md documentation with details for end user.

**Comment by hovsepm** [comment]
@src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Implementation\Features\GetAzureProviderFeatureCmdlet.cs, line 62 and 76 -> please add the string to project resources and use it from there.

**Comment by hovsepm** [comment]
in general: WriteWarning("The output object type of this cmdlet will be modified in a future release."); the text string should go to resources. please fix it everywhere.

**Comment by hovsepm** [comment]
src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Implementation\Resource\InvokeAzureResourceActionCmdlet.cs, line: 56 & 57, strings should go to resources as well

**Comment by hovsepm** [comment]
files in src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Implementation\ResourceGroupDeployments have very long lines which makes it hard to view on github, please limit them to 100~120 symbols per line.

**Comment by hovsepm** [comment]
files in src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Implementation\ResourceGroups have very long lines which makes it hard to view on github, please limit them to 100~120 symbols per line.

**Comment by hovsepm** [comment]
in src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Implementation\ResourceGroups\GetAzureResourceGroupCmdlet.cs, line 79. Instead of catching all the exceptions and silently continuing please write logs at debug stream - aka WriteDebug

**Comment by hovsepm** [comment]
src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\SdkClient\ResourceManagerSdkClient.cs -> add license header.

**Comment by hovsepm** [comment]
the naming convention for PS wrappers of model class is PS[nameOfTheModelToWrap]. you have several classes that violates that. Please fix:
* CreatePSResourceGroupDeploymentParameters.cs
* ValidatePSResourceGroupDeploymentParameters.cs
* CreatePSResourceGroupParameters.cs
* UpdatePSResourceGroupParameters.cs

**Comment by hovsepm** [comment]
src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Utilities\TemplateUtility.cs - missing license header.

**Comment by hovsepm** [comment]
src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Utilities\TemplateUtility.cs, line 108. Please add exception details through WriteDebug.

**Comment by hovsepm** [comment]
src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Utilities\TemplateUtility.cs, line 172. Change Debug.Assert to argument exception

**Comment by hovsepm** [comment]
due to huge changes please bump up major version in AzureRM.Resources.psd1

**Comment by hovsepm** [comment]
move strings to resources: 
* src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Implementation\ResourceGroupDeployments\NewAzureResourceGroupDeploymentCmdlet.cs line 69, 76
* src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Implementation\ResourceGroups\FindAzureResourceGroupCmdlet.cs line 78
* src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Implementation\ResourceGroups\NewAzureResourceGroupCmdlet.cs line 46
* \src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\Implementation\ResourceGroups\SetAzureResourceGroupCmdlet.cs line 54
* src\ResourceManager\Resources\Commands.ResourceManager\Cmdlets\SdkClient\ResourceManagerSdkClient.cs line 212, 213, 301, 595, 624, 784, 861, 865, 874,

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Properties/Resources.resx [hunk 7]
```diff
@@ -117,10 +117,238 @@
...
     <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
   </resheader>
+  <data name="AddingResourceGroup" xml:space="preserve">
+    <value>Adding resource group...</value>
+  </data>
+  <data name="CancelResourceGroupDeployment" xml:space="preserve">
+    <value>Are you sure you want to cancel the active deployment of resource group '{0}'?</value>
+  </data>
+  <data name="CancelResourceGroupDeploymentMessage" xml:space="preserve">
+    <value>Cancelling active resource group deployment ...</value>
```
Comment (gandhiniraj) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Properties/Resources.resx]: nit: no space after "deployment" and "..." or get line 121 in sync with this line

**Comment by hovsepm** [comment]
src\ResourceManager\Resources\Commands.Resources.Test\ScenarioTests\ResourcesController.cs, you should introduce an overload for RunPsTest which takes XunitTracingInterceptor as first parameter and then assigns it to helper.TracingInterceptor (e.g. https://github.com/Azure/azure-powershell/blob/dev/src/ResourceManager/Profile/Commands.Profile.Test/ProfileController.cs#L57). This will let you have output logs from PS code execution.

**Comment by vivsriaus** [comment]
@hovsepm I've addressed all the feedback, except for the last one: Adding the XunitTracingInterceptor to our test controller. I'm not comfortable making this change yet, as I don't know what the impact of this change will be on other RP projects that depend on this. If the above changes are merged, I will then be comfortable working on this tomorrow. Let me know if that sounds good.

**Comment by vivsriaus** [comment]
@hovsepm latest on demand job here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/847/

**Comment by vivsriaus** [comment]
@hovsepm latest sign job here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/439/

**Comment by vivsriaus** [comment]
Latest on demand here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/850/

**Comment by vivsriaus** [comment]
Updated on demand here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/853/

**Comment by hovsepm** [comment]
@vivsriaus "This branch has conflicts that must be resolved" please pull from upstream and resolve conflicts

**Comment by vivsriaus** [comment]
@hovsepm done. Latest on demand here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/854/

**Comment by vivsriaus** [comment]
@hovsepm things looking good on this pr?

**Comment by hovsepm** [comment]
@vivsriaus reviewing.

**Comment by hovsepm** [comment]
sign: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/857/

**Pull request merged**